src/main/c/cext/encoding.c

```diff
@@ -246,6 +246,16 @@ int rb_enc_precise_mbclen(const char *p, const char *e, rb_encoding *enc) {
   return polyglot_as_i32(RUBY_CEXT_INVOKE_NO_WRAP("rb_enc_precise_mbclen", rb_tr_temporary_native_string(p, length, enc)));
 }
 
+long rb_enc_strlen(const char *p, const char *e, rb_encoding *enc) {
+  long length = e - p;
+
+  if (rb_enc_mbminlen(enc) == rb_enc_mbmaxlen(enc)) {
+    return length / rb_enc_mbminlen(enc) + !!(length % rb_enc_mbminlen(enc));
```

Comment:
It seems a bit suboptimal to call `rb_enc_mbminlen(enc)` 3 times (and the C compiler can't deduplicate that), so I'll save the result in a local variable.